### PR TITLE
add items from fridge functionality to homepage

### PR DESCRIPTION
**Pull request by fransan6**
Need to refactor and sort out view but added functionality to include ingredients from fridge/kitchen to the search

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -18,4 +20,25 @@ def show
...
     @current_user_bookmarks = Bookmark.where(user: current_user)
   end
+
+  private
+
+  def user_ingredients(array)
+    array.map! { |ingredient_id| Ingredient.find(ingredient_id).name }.join(" ")
+    if params.values.include?("1")
+      checked = params.select { |_key, value| value == "1" }.keys
+      @recipes = Recipe.search_recipes("#{params[:query]} #{str}").where(checked.to_h { |value| [value, true] })
```
Comment (jondedman) [inline app/controllers/recipes_controller.rb]: Just wondering, where does the interpolates str variable come from. Is it defined elsewhere?

**Comment by jondedman** [comment]
Fran - this looks great. It took me a while to understand everything. I think i do now, but it might be good if you could talk me and Alex through it as we were not so involved with this bit. I think the only bit i didn't get was the part about the interpolated str variable. You are spot on that the form also works just fine without the turbo method. I'm happy to merge once I understand that str variable bit. I don't know if Alex also want to have a look as well though? Do we need to remove the link to the chat from the navbar, or shall we put that back in after the mid-way demo? I'm also thinking it would be cool if on the home page the link to the chat appeared after the recipes were filtered. It could say something like "See if your friend's have any of your missing ingredients". Probably, for after Thursday though.

File: app/controllers/recipes_controller.rb (hunk 1 above)
Comment (fransan6) [reply app/controllers/recipes_controller.rb]: Thanks for flagging - I didn't update that! Will sort it when I'm home and refactor a little as it needs work..

**Comment by fransan6** [comment]
Refactored the code so there's less of it and it's hopefully more readable. Should work for multiple instances of how a user might search but please do let me know if I've missed something! @alexagodzilla , I've included the functionality for if a user doesn't enter any ingredients but ticks a dietary filter.

**Comment by fransan6** [comment]
> Do we need to remove the link to the chat from the navbar, or shall we put that back in after the mid-way demo? I'm also thinking it would be cool if on the home page the link to the chat appeared after the recipes were filtered. It could say something like "See if your friend's have any of your missing ingredients". Probably, for after Thursday though.

Also, not sure about the chat question - guess that depends on if Alessandro and Ilaria get that done in time. I like the idea of having a little 'See if your friends have...' and adding that after Thursday. As a user can't see ingredients until they click on a recipe, perhaps the recipe show page might work better?

**Comment by ilariaperoni** [comment]
LGTM, thanks!!

**Pull request merged**